packages/credential-providers/src/fromAwsCliV2CompatibleProviderChain.ts

```diff
@@ -0,0 +1,77 @@
+import { remoteProvider } from "@aws-sdk/credential-provider-node/src/remoteProvider";
+import { createCredentialChain } from "@aws-sdk/credential-providers";
+import type { RuntimeConfigAwsCredentialIdentityProvider } from "@aws-sdk/types";
+import type { AwsCredentialIdentity } from "@aws-sdk/types";
+import { CredentialsProviderError } from "@smithy/property-provider";
+
+interface AwsCliV2CompatibleProviderOptions extends Partial<AwsCredentialIdentity> {
```

Comment:
this type is part of the public API and thus needs:

- the tsdoc `@public` annotation
- a description for each field (tsdoc)
- to be exported from the package